odoo/addons/base/res/res_font.py

```diff
@@ -52,57 +52,15 @@ class ResFont(models.Model):
         ('name_font_uniq', 'unique(family, name)', 'You can not register two fonts with the same name'),
     ]
 
-    @api.model
-    def font_scan(self, lazy=False):
-        """Action of loading fonts
-        In lazy mode will scan the filesystem only if there is no founts in the database and sync if no font in CustomTTFonts
-        In not lazy mode will force scan filesystem and sync
-        """
-        if lazy:
-            # lazy loading, scan only if no fonts in db
-            fonts = self.search([('path', '!=', '/dev/null')])
-            if not fonts:
-                # no scan yet or no font found on the system, scan the filesystem
-                self._scan_disk()
-            elif len(customfonts.CustomTTFonts) == 0:
-                # CustomTTFonts list is empty
-                self._sync()
-        else:
-            self._scan_disk()
-        return True
-
-    def _scan_disk(self):
-        """Scan the file system and register the result in database"""
-        found_fonts = []
-        for font_path in customfonts.list_all_sysfonts():
-            try:
-                font = ttfonts.TTFontFile(font_path)
-                _logger.debug("Found font %s at %s", font.name, font_path)
-                found_fonts.append((font.familyName, font.name, font_path, font.styleName))
-            except Exception, ex:
-                _logger.warning("Could not register Font %s: %s", font_path, ex)
-
-        for family, name, path, mode in found_fonts:
-            if not self.search([('family', '=', family), ('name', '=', name)]):
-                self.create({'family': family, 'name': name, 'path': path, 'mode': mode})
-
-        # remove fonts not present on the disk anymore
-        existing_font_names = [name for (family, name, path, mode) in found_fonts]
-        # Remove inexistent fonts
-        self.search([('name', 'not in', existing_font_names), ('path', '!=', '/dev/null')]).unlink()
-
-        self.pool.signal_caches_change()
-        return self._sync()
-
     def _sync(self):
         """Set the customfonts.CustomTTFonts list to the content of the database"""
-        customfonts.CustomTTFonts = []
+        CustomTTFonts = []
```

Comment:
That's going to create a new local variable and set it to an empty list. You need to declare `CustomTTFonts` as a global if you're going to *set* it. Same issue in clear_caches.